lib/discourse_plugin_registry.rb

```diff
@@ -246,8 +246,39 @@ def self.core_asset_for_name(name)
     asset
   end
 
+  def self.clear_filters!
+    @filters = nil
+  end
+
+  def self.register_filter(plugin_instance, name, &blk)
+    @filters ||= {}
+    filters = @filters[name] ||= []
+    filters << [plugin_instance, blk]
+  end
+
+  def self.apply_filter(name, arg, *more_args)
+    return arg if !@filters
+
+    filters = @filters[name]
```

Comment:
A tiny nitpick, `filters` clashes too much with the instance var in my mind, `registered_filters` might be better,  but really not a huge deal